gennotes_server/management/commands/add_all_clinvar_variants.py

```diff
@@ -1,26 +1,84 @@
+import codecs
+import fileinput
 from ftplib import FTP
 import gzip
+import json
+import md5
+from optparse import make_option
 import os
 import re
 import shutil
 import tempfile
 
-import reversion
-
 from django.contrib.auth import get_user_model
 from django.core.management.base import BaseCommand, CommandError
-from django.db import transaction
+from django.db import transaction, connection
+import reversion
+from vcf2clinvar.clinvar import ClinVarVCFLine
 
-from gennotes_server.models import Variant
+from gennotes_server.models import Variant, Relation
 from gennotes_server.utils import map_chrom_to_index
 
+
+try:
+    # faster implementation using bindings to libxml
+    from lxml import etree as ET
+except ImportError:
+    print 'Falling back to default ElementTree implementation'
+    from xml.etree import ElementTree as ET
+
 CV_VCF_DIR = 'pub/clinvar/vcf_GRCh37'
+CV_XML_DIR = 'pub/clinvar/xml'
+
 CV_VCF_REGEX = r'^clinvar_[0-9]{8}.vcf.gz$'
+CV_XML_REGEX = r'^ClinVarFullRelease_[0-9]{4}-[0-9]{2}.xml.gz$'
+
+SPLITTER = re.compile('[,|]')
+
+HASH_KEYS = ("type", "clinvar-accession:id", "clinvar-accession:version",
+             "clinvar-accession:significance",
+             "clinvar-accession:disease-name",
+             'num_submissions', 'record_status',
+             'gene:name', 'gene:symbol', 'citations')
 
 
 class Command(BaseCommand):
     help = 'Download latest ClinVar VCF, import variants not already in db.'
 
+    option_list = BaseCommand.option_list + (
+        make_option('-c', '--local-vcf',
+                    dest='local_vcf',
+                    help='Open local ClinVar VCF file'),
+        make_option('-x', '--local-xml',
+                    dest='local_xml',
+                    help='Open local ClinVar XML file'),
+        )
+
+    def _hash_xml_dict(self, d):
+        return md5.md5(json.dumps(
+            zip(HASH_KEYS, map(lambda k: d[k], HASH_KEYS)))).hexdigest()
```

Comment:
This died during running the command on the full dataset for me because there wasn't a "gene:name" available. I think maybe fixed by replacing `zip(HASH_KEYS, map(lambda k: d[k], HASH_KEYS))` with `zip(HASH_KEYS, map(lambda k: d[k] if k in d else '', HASH_KEYS))`

I tend to like writing these as list comprehensions fwiw, feels more readable for me :) `[(k, d[k] if k in d else '') for k in HASH_KEYS]`